### PR TITLE
Validator status improvements and fix

### LINKED ISSUE #5720: Validator status DEPOSITED never returned to Prometheus metrics
# 🐞 Bug Report

### Description

When adding a new validator key to my validator client, I was expecting my grafana dashboard to update my validator status from 0 (UNKNOWN) to 1 (DEPOSITED), then 2 (PENDING)

## 🔬 Minimal Reproduction

Create a new validator key pair, make deposit, and check for the validator client metrics to see that the value 1 (DEPOSITED) will never be returned

## 🔥 Error

The expected behavior is to get the status updated from 0 to 1 at the same time as in the log of the validator client. Instead I never saw the 1 status and it went directly to 2, skipping then the 1 status.

Both screeshot have been made at the exact same timing, just after restarting my validator client to make sure that my validator status was still to 0 or already at 1.
Validator client shows status at 1
https://cdn.discordapp.com/attachments/476588476393848832/706477013375909908/unknown.png

Prometheus metrics shows status at 0
https://cdn.discordapp.com/attachments/476588476393848832/706477312379584512/unknown.png

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Improvement 

**What does this PR do? Why is it needed?**
This PR changes all references of `ValidatorAtIndex` to `ValidatorAtIndexReadOnly` since there are no modifications made to validators throughout this function.

Also fixes a reporting issue made in #5396 

**Which issues(s) does this PR fix?**
Fixes #5720

### CODE REVIEW

File: beacon-chain/cache/depositcache/deposits_cache.go [hunk 1]
```diff
@@ -162,7 +162,10 @@ func (dc *DepositCache) DepositByPubkey(ctx context.Context, pubKey []byte) (*et
 
 	var deposit *ethpb.Deposit
 	var blockNum *big.Int
-	for _, ctnr := range dc.deposits {
+	// Search the list backwards since this function is only used for
+	// pending activation validators and the newest deposits are always at the end.
+	for i := len(dc.deposits) - 1; i > 0; i-- {
```
Comment (0xKiwi) [inline beacon-chain/cache/depositcache/deposits_cache.go:167]: If anyone disagrees with this change please let me know, this function is only used for newer validators so I figured it'd make sense.

Comment (rauljordan) [reply beacon-chain/cache/depositcache/deposits_cache.go:167]: This makes sense, nice catch!

**Comment by 0xKiwi** [comment]
@rauljordan it seems like the deposit change breaks a test `TestBeaconDB_DepositByPubkey_ReturnsFirstMatchingDeposit`. Is there a strong reason for looking for the first deposit?

File: beacon-chain/cache/depositcache/deposits_cache.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/cache/depositcache/deposits_cache.go:167]: Please change this back, this picks the wrong deposit in the event of top-ups. Its why the tests fail now.

**Pull request merged**